src/modules/delete.js

```diff
@@ -5,15 +5,19 @@ export let tasks = JSON.parse(localStorage.getItem('tasks')) || [];
 export default function deleteTask() {
   const deleteBtn = document.querySelectorAll('.option');
   deleteBtn.forEach((btn) => {
-    const container = document.querySelector('.container');
     btn.addEventListener('click', (e) => {
+      const deletedTaskIndex = e.target.dataset.index;
       const item = e.target.parentNode.parentNode;
-      container.removeChild(item);
-      const deletedTask = e.target.previousElementSibling.textContent;
-      tasks = tasks.filter((task) => task.description !== deletedTask);
-      localStorage.setItem('tasks', JSON.stringify(tasks));
+      item.remove();
+      tasks.splice(deletedTaskIndex, 1);
+
+      // update index of remaining tasks
+      tasks.forEach((task, i) => {
```

Comment:
- [ ] Kidly make sure when the delete icon is pressed, the task is deleted from the DOM and the local storage as well.
`Hint: You may consider say:`
```js
// When the button is clicked you reset the array of tasks by filtering them
//   excluding the one which was deleted and 
 //  then you update the list in your localStorage 
  //  by setting this new array that exclude the deleted task
```